blob/gcsblob/gcsblob.go

```diff
@@ -59,46 +59,48 @@ import (
 const defaultPageSize = 1000
 
 func init() {
-	blob.Register("gs", func(ctx context.Context, u *url.URL) (driver.Bucket, error) {
-		q := u.Query()
-		opts := &Options{}
+	blob.Register("gs", openURL)
+}
 
-		if accessID := q["access_id"]; len(accessID) > 0 {
-			opts.GoogleAccessID = accessID[0]
-		}
+func openURL(ctx context.Context, u *url.URL) (driver.Bucket, error) {
+	q := u.Query()
+	opts := &Options{}
 
-		if keyPath := q["private_key_path"]; len(keyPath) > 0 {
-			pk, err := ioutil.ReadFile(keyPath[0])
-			if err != nil {
-				return nil, err
-			}
-			opts.PrivateKey = pk
-		}
+	if accessID := q["access_id"]; len(accessID) > 0 {
```

Comment:
nit: `if accessID, ok := q["access_id"]; ok {`

Reply:
Hmm, I think checking that `len > 0` is more correct here; technically the map could have an `access_id` key with a 0-length slice value, and `ok` would be `true` but the access of `[0]` on the next line would panic. It's possible that `net.URL` would never produce a 0-length slice value, but I think its better to check the thing we actually care about.